newton/_src/sim/ik/ik_lbfgs_optimizer.py

```
@@ -0,0 +1,1623 @@
+
+        @wp.kernel(module="unique")
+        def _fk_local(
+            joint_type: wp.array1d(dtype=wp.int32),  # (n_joints)
+            joint_q: wp.array2d(dtype=wp.float32),  # (n_problems, n_coords)
+            joint_q_start: wp.array1d(dtype=wp.int32),  # (n_joints + 1)
+            joint_qd_start: wp.array1d(dtype=wp.int32),  # (n_joints + 1)
+            joint_axis: wp.array1d(dtype=wp.vec3),  # (n_axes)
+            joint_dof_dim: wp.array2d(dtype=wp.int32),  # (n_joints, 2)  → (lin, ang)
+            joint_X_p: wp.array1d(dtype=wp.transform),  # (n_joints)
+            joint_X_c: wp.array1d(dtype=wp.transform),  # (n_joints)
+            joint_count: int,
```

Ruff (0.13.1)

Unused function argument: `joint_count`

(ARG001)